Makefile

```diff
@@ -97,18 +102,30 @@ race: deps ## Run data race detector
 
 .PHONY: coverage
 coverage: deps ## Generate code coverage report
-	@go test $(UNIT_TESTS) -v -coverprofile coverage.cov
+	@mkdir -p $(GOCOVERDIR)
+	@go test $(UNIT_TESTS) -v -coverprofile $(GOCOVERDIR)/coverage_output.cov -covermode atomic
 	# exclude mock files from coverage
-	@cat coverage.cov | grep -v _mock.go > coverage_cleaned.cov || true
-	@go tool cover -func=coverage_cleaned.cov
+	@cat $(GOCOVERDIR)/coverage_output.cov | grep -v _mock.go > $(GOCOVERDIR)/coverage.cov || true
+	@rm $(GOCOVERDIR)/coverage_output.cov
+	@go tool cover -func=$(GOCOVERDIR)/coverage.cov
 
 .PHONY: coverhtml
 coverhtml: coverage ## Generate HTML coverage report
-	@go tool cover -html=coverage_cleaned.cov -o coverage_unit.html
+	@go tool cover -html=$(GOCOVERDIR)/coverage.cov -o $(GOCOVERDIR)/coverage_unit.html
 
 deploy/dockerfiles: ## Clone Dockerfiles repository
 	@git clone git@github.com:$(REPOSITORY_OWNER)/dockerfiles.git deploy/dockerfiles
 
+.PHONY: run-with-coverage
+run-with-coverage: build-cover ## Run binary and capture code coverage (during e2e tests)
+	@mkdir -p $(GOCOVERDIR)
+	@GOCOVERDIR=$(GOCOVERDIR) ./$(PROJECT_NAME)
+
+.PHONY: convert-coverage-to-text
```

Comment:
```suggestion
.PHONY: convert-run-coverage
```

Reply:
Thanks, you identified a leftover from a refactoring I did 🤓